src/pycolmap/geometry/bindings.cc

```diff
@@ -85,7 +83,6 @@ void BindGeometry(py::module& m) {
       .def_readwrite("translation", &Rigid3d::translation)
       .def("matrix", &Rigid3d::ToMatrix)
       .def("adjoint", &Rigid3d::Adjoint)
-      .def("essential_matrix", &EssentialMatrixFromPose)
```

Comment:
From a user standpoint, what is the rational for removing this method? It does not introduce annoying dependencies and it does not clutter the API. I suggest to keep this method in addition to the new binding in `geometry/essential_matrix.cc`.

Reply:
It does introduce a dependency from Rigid3d on essential matrix functionality and, semantically, not every Rigid3d is a meaningful essential matrix. If we are convinced about having it bound as a method to Rigid3d, we should likewise also introduce it on the C++ side? I understand this introduces a slight degradation in convenience when doing this conversion, but from a design perspective it is the cleaner and more consistent API. If you prefer, I can trigger an API deprecation warning instead?

Reply:
Keeping the C++ and Python API consistent indeed seems important. I don't have a strong opinion about this, we can remove the method without deprecation warning. 

Alternatively, we could move `EssentialMatrixFromPose`, `InterpolateCameraPoses` to rigid3.h and `TransformCameraWorld` to sim3.h to make the C++ & Python even more consistent.

Reply:
> Alternatively, we could move EssentialMatrixFromPose, InterpolateCameraPoses to rigid3.h and TransformCameraWorld to sim3.h to make the C++ & Python even more consistent.

I would keep them where they are right now. I currently like that rigid3d and sim3d are useful independent of colmap and projective geometry. One could copy it without changes into another project that needs euclidean/metric coordinate transformations.